### ISSUE TITLE
Link to (remote) module docs from module `source`

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/vscode-terraform#673]
## Problem Statement

The Terraform Registry automatically generates documentation for module authors from `Readme.md` and variables and outputs.

It would be helpful for users to be able to easily get to the overview/documentation of a module from the source code.

## Expected User Experience

```hcl
module "consul" {
  source = "hashicorp/consul/aws"
  version = "0.1.0"
}
```

User is taken to the docs at https://registry.terraform.io/modules/hashicorp/consul/aws/0.1.0 upon `ctrl/cmd+click`.

The UI is similar to the one we use for provider docs currently:
![Screenshot 2021-06-18 at 17 06 57](https://user-images.githubusercontent.com/287584/122588981-9d04c600-d057-11eb-81d8-fcf64cd581de.png)

## Proposal / TODO

 - [x] https://github.com/hashicorp/hcl-lang/pull/115
 - [x] https://github.com/hashicorp/terraform-registry-address/pull/7
 - [x] https://github.com/hashicorp/terraform-schema/pull/103
 - [x] https://github.com/hashicorp/terraform-ls/pull/874
 - [x] https://github.com/microsoft/vscode/issues/147447

**Comment by vide** [comment]
It might be unrelated but with the older, unofficial Terraform VSCode extension you could ctrl+click a resource name and get the help page for that resource on terraform.io/docs automatically. I'm not able anymore to achieve that, is it a missing feature or am I doing something wrong?

**Comment by radeksimko** [comment]
This is being tracked under https://github.com/hashicorp/terraform-ls/issues/190 - we don't have a corresponding issue in this repo (yet) but all of the work will be done on the language server anyway.

Feel free to subscribe to that issue.

**Comment by dbanck** [comment]
Resolved via https://github.com/hashicorp/terraform-ls/issues/874 and will be included in the [0.28.0](https://github.com/hashicorp/terraform-ls/milestone/20) language server release.
